perun-base/src/main/java/cz/metacentrum/perun/core/api/Member.java

```diff
@@ -128,7 +128,7 @@ public boolean equals(Object obj) {
 		if (this == obj) {
 			return true;
 		}
-		if (obj == null) {
+		if (!(obj instanceof Member)) {
```

Comment:
Shouldn't this be at line 134 ? Replacing ``.getClass()`` check instead of ``obj == null`` check ? Like in ``User`` object ? I know it will return ``false`` for ``null`` anyway, but this will also return false if you compare ``Member`` and ``RichMember`` which we wanted to be comparable. In ``User`` object you just continue with params comparison when you compare ``User`` and ``Candidate`` or ``RichUser``. I believe implementation should be at least equivalent in both ``Member`` and ``User`` objects.

Reply:
You are right. It should be fixed now.